commit f42ebb40a8cbee27e04e81113d6fd15964a903b2 (refs/changes/17/117/2)
Author: Pedro P. Candel <pedro@joyent.com>
Date:   2016-07-21T13:41:56+02:00 (3 years, 3 months ago)
    
    TOOLS-1492 `sdcadm insts` exceptions on SAPI instances for removed servers

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index e261af1..34aeb41 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -815,13 +815,21 @@ SdcAdm.prototype.listInsts = function listInsts(opts, cb) {
                             if (ins.params && ins.params.server_uuid) {
                                 var id = ins.params.server_uuid;
                                 d.server = id;
+                                // TOOLS-1492: A server may have been deleted
+                                // leaving the SAPI instance present and
+                                // associated to nothing:
+                                if (!ctx.serverFromUuid[id]) {
+                                    return null;
+                                }
                                 d.hostname = ctx.serverFromUuid[id].hostname;
                                 d.server_ip = ctx.serverAdminIpFromUuid[id];
                             }
                             return (d);
+                        }).filter(function (ins) {
+                            return (ins !== null);
                         });
                         ctx.insts = ctx.insts.concat(dlInsts);
-                        return next();
+                        next();
                     });
                 });
             });
